pkg/storage/image.go

```diff
@@ -597,16 +598,47 @@ func GetImageService(ctx context.Context, store storage.Store, defaultTransport
 		seenRegistries[r] = true
 	}
 
+	// support global shared /etc/containers/registries.conf
+	allSystemRegistries, err := sysregistriesv2.GetRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+	insecureSystemRegistries := []string{}
+	for _, r := range allSystemRegistries {
+		if r.Insecure {
+			insecureSystemRegistries = append(insecureSystemRegistries, r.URL)
+		}
+	}
```

Comment:
we still want the insecure registries configuration from crio.conf to take precedence for backwards compatibility (or am I missing something?)

Reply:
- The PR, right now, says
   ```go
   if len(crioInsecureRegistries) != 0 {
       /* Among other effects, also essentially ignore insecureSystemRegistries */
   }
   ```
   Removing `insecureSystemRegistries`, because c/image/docker handles them already, makes this behavior/concern go away.
- (BTW the way “insecure registries” work, where a registry can be marked as insecure but not as secure, there does not have to be a precedence at all — we can use a union of all “marked as insecure” configurations; and that’s what happens when this code only cares about `crioInsecureRegistries` and c/image/docker handles `sysregistriesv2`).